packages/backend/src/ee/services/ai/tools/findExplores.ts

```diff
@@ -1,51 +1,145 @@
+        </Dimensions>
+
+        <Metrics alt="metric ids" size="${metrics.length}">
+            ${metrics
+                .map(
+                    (metric) =>
+                        `<Metric fieldId="${getItemId({
+                            name: metric.name,
+                            table: metric.tableName,
+                        })}" popularity="${metric.chartUsage ?? 0}">${
+                            metric.label
+                        }</Metric>`,
+                )
+                .join('\n')}
+        </Metrics>
+    </Fields>
+</Explore>`.trim();
+};
+
+export const getFindExplores = ({ findExplores }: Dependencies) => {
+    const schema = z.object({
+        page: z
+            .number()
+            .nullable()
+            .describe(
+                'Use this to paginate through the results. Starts at 1 and increments by 1.',
+            ),
+    });
 
     return tool({
-        description: `Use this tool to get information about Explores (models) available to you.`,
+        description: `Use this tool to list available Explores with joined tables with hints, description, and available fields. The list is paginated, you can paginate through the results by recalling the tool with the next page token.`,
         parameters: schema,
-        execute: async () => {
+        execute: async (args) => {
             try {
-                const explores = await getExplores();
-
-                return `Here is the information about the explores/models you have access to:
-${explores
-    .map(
-        (explore) => `Explore/Model id: ${explore.name}
-Name/Label: ${explore.label}
-${
-    explore.aiHint
-        ? `Hints:\n${convertToAiHints(explore.aiHint)
-              ?.map((hint) => `- ${hint}`)
-              .join('\n')}`
-        : ''
-}
-Description: ${explore.description ?? 'No description'}
-${
-    explore.joinedTables.length > 0
-        ? `Base Table: ${explore.baseTable}
-Joined tables: ${explore.joinedTables.join(', ')}`
-        : ''
-}
-`,
-    )
-    .join('\n\n--------------------------------\n\n')}`;
+                if (args.page && args.page < 1) {
+                    return `Error: Page must be greater than 0.`;
+                }
+
+                const { pagination, fields, tables } = await findExplores({
+                    page: args.page ?? 1,
+                    pageSize: PAGE_SIZE,
+                });
+
+                const exploreResponses = tables
+                    .map((table) => generateExploreResponse(table, fields))
+                    .join('\n\n');
+
+                return `<Explores page="${pagination?.page}" pageSize="${pagination?.pageSize}" totalPageCount="${pagination?.totalPageCount}" totalResults="${pagination?.totalResults}">
+                ${exploreResponses}
+</Explores>`;
             } catch (error) {
-                return toolErrorHandler(error, `Error finding explores.`);
+                return toolErrorHandler(error, `Error listing explores.`);
```

Comment:
think this can be removed now

Reply:
This is a generic error handler we have for each tool. It guarantees that if something errors, it loops back in the agent call and the agent self-corrects if it is a recoverable error.